### PR TITLE
miniz: 2.2.0 → 3.0.2

### PR DESCRIPTION
**Pull request by nicoonoclaste**
## Description of changes

In `miniz`:
- update from 2.2.0 to 3.0.2,
- add `updateScript`,
- add `meta.pkgConfigModules` and `tests.pkg-config`,
- add the `validatePkgConfig` hook.

cc @astro

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested, as applicable:
  - new [package test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
- [ ] Tested compilation of all packages that depend on this change using `nixpkgs-review`.
  Currently running
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/miniz/default.nix [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
+  version = "3.0.2";
 
   src = fetchFromGitHub {
     owner = "richgel999";
-    repo = pname;
-    rev = version;
-    sha256 = "sha256-7hc/yNJh4sD5zGQLeHjowbUtV/1mUDQre1tp9yKMSSY=";
+    repo = finalAttrs.pname;
+    rev = finalAttrs.version;
+    sha256 = "sha256-3J0bkr2Yk+MJXilUqOCHsWzuykySv5B1nepmucvA4hg=";
```
Comment (AndersonTorres) [inline pkgs/development/libraries/miniz/default.nix:18]:
```suggestion
    hash = "sha256-3J0bkr2Yk+MJXilUqOCHsWzuykySv5B1nepmucvA4hg=";
```

File: pkgs/development/libraries/miniz/default.nix [hunk 2]
```diff
@@ -1,29 +1,40 @@
...
   };
+  passthru.updateScript = nix-update-script {};
```
Comment (AndersonTorres) [inline pkgs/development/libraries/miniz/default.nix:20]:
```suggestion
```

File: pkgs/development/libraries/miniz/default.nix [hunk 3]
```diff
@@ -1,29 +1,40 @@
...
   postFixup = ''
-    substituteInPlace "$out"/share/pkgconfig/miniz.pc \
-      --replace '=''${prefix}//' '=/' \
-      --replace '=''${exec_prefix}//' '=/'
+    substituteInPlace "$out"/lib/pkgconfig/miniz.pc \
+      --replace-fail '=''${prefix}//' '=/' \
+      --replace-fail '=''${exec_prefix}//' '=/'
   '';
 
+  passthru.tests.pkg-config = testers.testMetaPkgConfig finalAttrs.finalPackage;
```
Comment (AndersonTorres) [inline pkgs/development/libraries/miniz/default.nix:30]:
```suggestion
  passthru = {
    updateScript = nix-update-script { };
    tests = {
      pkg-config = testers.testMetaPkgConfig finalAttrs.finalPackage;
    };
  };
```

File: pkgs/development/libraries/miniz/default.nix [hunk 4]
```diff
@@ -1,29 +1,40 @@
...
+
   meta = with lib; {
     description = "Single C source file zlib-replacement library";
     homepage = "https://github.com/richgel999/miniz";
     license = licenses.mit;
     maintainers = with maintainers; [ astro ];
     platforms = platforms.unix;
```
Comment (AndersonTorres) [inline pkgs/development/libraries/miniz/default.nix:37]:
```suggestion
  meta = {
    description = "Single C source file zlib-replacement library";
    homepage = "https://github.com/richgel999/miniz";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ astro ];
    platforms = lib.platforms.unix;
```

File: pkgs/development/libraries/miniz/default.nix [hunk 5]
```diff
@@ -1,29 +1,40 @@
-{ lib, stdenv, fetchFromGitHub, cmake }:
+{ lib
+, fetchFromGitHub
+, nix-update-script
+, stdenv
+, testers
+, validatePkgConfig
+, cmake
+}:
```
Comment (AndersonTorres) [inline pkgs/development/libraries/miniz/default.nix:8]:
```suggestion
{
  lib,
  cmake,
  fetchFromGitHub,
  nix-update-script,
  stdenv,
  testers,
  validatePkgConfig,
}:
```

New formatter rules.

File: pkgs/development/libraries/miniz/default.nix [hunk 6]
```diff
@@ -1,29 +1,40 @@
...
   pname = "miniz";
-  version = "2.2.0";
+  version = "3.0.2";
 
   src = fetchFromGitHub {
     owner = "richgel999";
-    repo = pname;
-    rev = version;
-    sha256 = "sha256-7hc/yNJh4sD5zGQLeHjowbUtV/1mUDQre1tp9yKMSSY=";
+    repo = finalAttrs.pname;
```
Comment (AndersonTorres) [inline pkgs/development/libraries/miniz/default.nix:16]:
```suggestion
    repo = "miniz";
```

File: pkgs/development/libraries/miniz/default.nix [hunk 7]
```diff
@@ -1,29 +1,40 @@
...
+    rev = finalAttrs.version;
+    sha256 = "sha256-3J0bkr2Yk+MJXilUqOCHsWzuykySv5B1nepmucvA4hg=";
   };
+  passthru.updateScript = nix-update-script {};
 
-  nativeBuildInputs = [ cmake ];
+  nativeBuildInputs = [ cmake validatePkgConfig ];
```
Comment (AndersonTorres) [inline pkgs/development/libraries/miniz/default.nix:23]:
```suggestion

  strictDeps = true;

```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/miniz/default.nix (hunk 4 above)
Comment (AndersonTorres) [reply pkgs/development/libraries/miniz/default.nix:37]: - Tracking issue:  https://github.com/NixOS/nixpkgs/issues/208242
- Best Practices: https://nix.dev/guides/best-practices#with-scopes
- A real-life bug: https://github.com/NixOS/nixpkgs/pull/295975
- A rather complicated real-life bug: https://github.com/NixOS/nixpkgs/issues/292468#issuecomment-2016454242

File: pkgs/development/libraries/miniz/default.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/development/libraries/miniz/default.nix:30]: - "Keeping next" is not relevant. This is not an imperative language, the order of attributes does not matter.
- Scattered `passthru` attributes is strictly less legible.
- Hammering Guide: https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/attribute-ordering.md

----------

- I concede the unfurling point. My intention is that usually other tests can be added in the future, but indeed it is less legible as it stands now.

**Review by mkg20001 — APPROVED** [review]
change looks good, passes formatting ci

File: pkgs/development/libraries/miniz/default.nix (hunk 7 above)
Comment (AndersonTorres) [reply pkgs/development/libraries/miniz/default.nix:23]: https://nixos.org/manual/nixpkgs/stable/#ssec-stdenv-dependencies-reference

Comment (AndersonTorres) [reply pkgs/development/libraries/miniz/default.nix:24]: The reference was removed, I still don't know why. I will search the commit logs.

Summarizing a bit, `strictDeps =  true` enforces a clearer separation between the various `*buildInputs`. It is helpful e.g. in cross-compilation and porting to other environments.

```
 By default, the run-time/build-time distinction is just a hint for mental clarity, but with strictDeps set it is mostly enforced even in the native case.
```

**Pull request merged**